modal/_utils/blob_utils.py

```diff
@@ -412,12 +421,13 @@ def get_file_upload_spec_from_path(
     # Python appears to give files 0o666 bits on Windows (equal for user, group, and global),
     # so we mask those out to 0o755 for compatibility with POSIX-based permissions.
     mode = mode or os.stat(filename).st_mode & (0o7777 if platform.system() != "Windows" else 0o7755)
-    return _get_file_upload_spec(
-        lambda: open(filename, "rb"),
-        filename,
-        mount_filename,
-        mode,
-    )
+    if os.path.getsize(filename) > MULTIPART_UPLOAD_THRESHOLD:
```

Comment:
Can't we do this size check in the same place that we check for LARGE_FILE_LIMIT, in `_get_file_upload_spec`?
`_get_file_upload_spec` supports BinaryIO objects as import as well, and it feels like this fix could be beneficial for those too?

Reply:
how can we check the length of a BinaryIO object?

Reply:
Let me take a look.

Reply:
Done!